compiler/tests/fixtures/checked_arithmetic.fe

```diff
@@ -70,4 +70,45 @@ contract CheckedArithmetic:
         return left - right
 
     pub def sub_i8(left: i8, right: i8) -> i8:
-        return left - right
\ No newline at end of file
+        return left - right
+
+    pub def mul_u256(left: u256, right: u256) -> u256:
+        return left * right
+
+    pub def mul_u128(left: u128, right: u128) -> u128:
+        return left * right
+
+    pub def mul_u64(left: u64, right: u64) -> u64:
+        return left * right
+
+    pub def mul_u32(left: u32, right: u32) -> u32:
+        return left * right
+
+    pub def mul_u16(left: u16, right: u16) -> u16:
+        return left * right
+
+    pub def mul_u8(left: u8, right: u8) -> u8:
+        return left * right
+
+    pub def weird_bug() -> i256:
+        val1: i256 = i256(-57896044618658097711785492504343953926634992332820282019728792003956564819968)
+        val2: i256 = i256(1)
+        return val1 * val2
```

Comment:
This is `min_val_i256 * 1` and it fails and I have no idea why. The same thing works in solidity.